src/antsibull_nox/config.py

```diff
@@ -173,6 +175,63 @@ class SessionLicenseCheck(_BaseModel):
     license_check_extra_ignore_paths: list[str] = []
 
 
+class ExecutionEnvironmentConfig(_BaseModel):
+    """
+    Execution enviroment check session config.
+    """
+
+    name: str
+    description: t.Optional[str] = None
+    test_playbooks: list[str]
+    version: int = 3
+    base_image_name: str = "quay.io/fedora:latest"
+    ansible_core_source: t.Literal["package_pip", "package_system"] = "package_pip"
+    ansible_core_package: str = "ansible-core"
+    ansible_runner_package: str = "ansible-runner"
+    system_packages: list[str] = []
+    python_packages: list[str] = []
+    python_interpreter_package: t.Optional[str] = None
+    python_path: t.Optional[str] = None
+
+    def to_execution_environment_config(self) -> dict[str, t.Any]:
+        """
+        Convert TOML config to execution environment YAML.
+        """
+
+        dependencies: dict[str, t.Any] = {}
+        dependencies["ansible_core"] = {
+            self.ansible_core_source: self.ansible_core_package
+        }
+        dependencies["ansible_runner"] = {"package_pip": self.ansible_runner_package}
+        if self.python_interpreter_package:
+            python_interpreter: dict[str, t.Any] = {
+                "package_system": self.python_interpreter_package
+            }
+            if self.python_path:
+                python_interpreter["python_path"] = self.python_path
+            dependencies["python_interpreter"] = python_interpreter
+        if self.system_packages:
+            dependencies["system"] = self.system_packages
+        if self.python_packages:
+            dependencies["python"] = self.python_packages
+
+        return create_ee_config(
+            version=self.version,
+            base_image=self.base_image_name,
+            dependencies=dependencies,
+        )
+
+
+class SessionExecutionEnvironmentCheck(_BaseModel):
+    """
+    Execution environment check session.
+    """
+
+    default: bool = False
+
+    execution_environments: dict[str, ExecutionEnvironmentConfig] = {}
```

Comment:
I would simply use a list here. Also I would remove the default, since not specifying anything doesn't really make sense.
```suggestion
    execution_environments: list[ExecutionEnvironmentConfig]
```